### PR TITLE
Faker source gains purchases and products

### PR DESCRIPTION
**Pull request by evantahler**
Part of https://github.com/airbytehq/airbyte/issues/13201

Screen Shot 2022-05-26 at 4 46 37 PM
Screen Shot 2022-05-26 at 4 48 27 PM

The faker command now has 3 streams: Users, Products, and Purchases.  The list of Products is a static list of 100 cars.  The list of purchases is generated randomly against certain percentages.  Each purchase also has 3 timestamps: added_to_cart, purchased, and returned - these are always in the proper chronological order, and happen with various percentages: 70% likely to move from cart -> purchase and 30% likely to return the item if it was purchased. 

This dataset was designed to allow for exploration and transformation with dbt and various reporting tools to produce data like "user lifetime value" and "avg rate of return". 

```sql
SELECT 
	users.id as user_id
	, mail as email
	, name
	, COUNT(purchases.*) as count_purchases
	, SUM(products.price) as ltv
from users
join purchases on users.id = purchases.user_id
join products on purchases.product_id = products.id
group by
	users.id, users.mail, users.name
order by ltv desc
;
```
 
There are also now additional configuration options to control the size of each read and stream-slice (page). This will prove helpful for debugging things (like checkpointing) down the road.

### CODE REVIEW

**Comment by evantahler** [comment]
cc @chrisroseairbyte

File: docs/integrations/sources/faker.md
Comment (Phlair) [inline docs/integrations/sources/faker.md:44]: this looks incomplete, just a reminder :)

**Review by Phlair — APPROVED** [review]
Awesome, who needs real data anyway!? 👍 

couple of nits and small suggestions so no need for a second review, approving with this one.

File: airbyte-integrations/connectors/source-faker/source_faker/source.py
Comment (evantahler) [reply airbyte-integrations/connectors/source-faker/source_faker/source.py:133]: dangit! I was trying `records_in_page++` and that didn't compile, so I gave up trying shorhands :p

File: airbyte-integrations/connectors/source-faker/source_faker/source.py
Comment (evantahler) [reply airbyte-integrations/connectors/source-faker/source_faker/source.py:160]: I think I need it do this doesn't fall into the failure ELSE case at the end

File: airbyte-integrations/connectors/source-faker/source_faker/source.py
Comment (evantahler) [reply airbyte-integrations/connectors/source-faker/source_faker/source.py:142]: good point! now it will throw :D

**Pull request merged**
